content/articles/rocket-lander-in-elm.md

```diff
@@ -0,0 +1,198 @@
+[Fay](https://github.com/faylang/fay/wiki),
+[clojurescript](https://github.com/clojure/clojurescript),
+[scala.js](http://www.scala-js.org/).
+
+
+## Enter Elm
+
+Elm is a statically typed, functional language with immutable data
+structures and first-class support for [functional reactive
+programming](http://en.wikipedia.org/wiki/Functional_reactive_programming).
+
+Elm also comes with a standard library that contains lots of useful
+tools such as an implementation of the
+[virtual-DOM](http://stackoverflow.com/questions/21109361/why-is-reacts-concept-of-virtual-dom-said-to-be-more-performant-than-dirty-mode),
+famous for for making [React](http://facebook.github.io/react/) so
+speedy.
+
+Like in Haskell, type annotations can be given inline, or in a line
+above a function, and types always start with upper case:
+
+```elm
+1 : Int
+True : Bool
+["ab", "cde"] : List String
+
+mul : Int -> Int -> Int
+mul a b = a * b
+```
+
+Elm starts execution at the `main` entry point:
+
+```
+import Text
+
+main = Text.plainText "Hello from Elm."
+```
+
+To see more code check out the rocket lander
+[on GitHub](https://github.com/WeAreWizards/elm-rocket-lander/blob/master/Ship.elm)
+
+To get going I recommend installing via Haskell's
+[Cabal-Install](https://www.haskell.org/cabal/) which is
+available on most distributions:
+
+```
+cabal update
+cabal install elm-make elm-package elm-compiler
+```
+
+This is not a tutorial so I refer to Elm's
+[own documentation](http://elm-lang.org/Learn.elm) to learn more.
+
+
+# FRP
+
+Elm's functional reactive programming (FRP) connects inputs
+("signals") to code that depends on these signals, forming a directed
+graph in the process.
+
+Anything that generates DOM events can be a signal in Elm. E.g. the
+mouse moving or a key press. Elm also has a signal that generates a
+steady tick, e.g. at 30 frames-per-second, which is useful for
+programming games.
+
+Every signal in Elm propagates to all code that depends on it. The
+flow is usually something like (Input -> Update application state ->
+Redraw application).
+
+Elm requires setting up the entire signal graph *before* the
+application starts. That makes it easier to reason about inputs
+flowing through the code, but it also comes with some downsides:
+
+E.g. for the
+[rocket lander demo](/rocket-lander-in-elm-extra/ship.html) I need a
+30 frames-per-second signal to animate the game. I have to set that up
+at game start and can't switch it off. I.e. the help-screen at the
+beginning is re-rendering 30 times a second, even though nothing
+changes.
+
+For a silly little Game that's probably fine. Consider though: An app
+that deals with user input will see a few events a second with long
+pauses of nothing happening. If you wanted to add an animation that
+fades out at 30 frames per second then the app would need to run at 30
+FPS the whole time. In these cases it's best to use JavaScript
+directly via Elm's ([ports](http://elm-lang.org/learn/Ports.elm)).
```

Comment:
Again: Are you making an argument against Elm, or against the paragraph as written?